scenario/openapp.go

```diff
@@ -304,3 +305,21 @@ func (connectWs connectWsSettings) Execute(sessionState *session.State, actionSt
 func (connectWs connectWsSettings) Validate() error {
 	return nil
 }
+
+// AffectsAppObjectsAction implements AffectsAppObjectsAction interface
+func (settings OpenAppSettings) AffectsAppObjectsAction(structure appstructure.AppStructure) (*appstructure.AppStructurePopulatedObjects, []string, bool) {
+	newObjs := appstructure.AppStructurePopulatedObjects{
+		Parent:    settings.App.String(),
+		Objects:   make([]appstructure.AppStructureObject, 0),
+		Bookmarks: nil,
+	}
+	for _, obj := range structure.Objects {
+		if obj.Type == "sheet" {
+			newObjs.Sheets = append(newObjs.Sheets, obj)
+		}
+	}
+	for _, v := range structure.Bookmarks {
+		newObjs.Bookmarks = append(newObjs.Bookmarks, v)
+	}
+	return &newObjs, nil, true
+}
```

Comment:
openapp should not add any active objects other than bookmarks

Reply:
It should add its sheets to the list of currently available sheets?

Reply:
changesheet should filter on all instead of on active.

Reply:
I'm not following the distinction. If there are multiple apps in a script, it should only show the sheets of the currently open app, i.e "active"?

Reply:
only the current app yes. But there's a list of "active objects" and a dictionary of "all objects". Instead of handling sheets separate, we need a way for an action to tell if the filter types, in the case of changesheet this would be `sheet` should be applied on the list of active objects (i.e. "added" by an action) or on the dictionary of all objects in the app.